### PR TITLE
Handle generators in IngesterManager

### PR DESCRIPTION
**Pull request by jgaff**
Handle generators returned from extensions in `IngesterManager.run_extensions()`

### CODE REVIEW

File: pif_ingestor/manager.py [hunk 1]
```diff
@@ -40,8 +43,13 @@ def run_extensions(self, files, args={}, include=None, exclude=[]):
             extension = self.extension_manager[name]
             try:
                 pifs = extension.plugin.convert(files, **args)
+                # Return value from convert can be System, list of Systems, or generator
+                if isinstance(pifs, System):
+                    pifs = [pifs]
+                else:
+                    pifs = [p for p in pifs]
```
Comment (maxhutch) [inline pif_ingestor/manager.py]: This will force all the pifs to be loaded into memory, defeating the purpose of the generator.

Comment (maxhutch) [reply pif_ingestor/manager.py]: Good point that our `auto` format logic relies on getting the whole result.  How about we force the whole generator into memory for `auto` but leave it un-collapsed if the format is explicit?

Comment (jgaff) [reply pif_ingestor/manager.py]: What happens if the user submits `include=[list of multiple extensions]`? The current logic is to just return the first run that has results, but if the first extension returns a generator we're back to the original problem. If the user has exactly one extension to run, `run_extension()` (singular) is a better(?) option. Is is acceptable to ask users to run `run_extension()` once per extension, if their chosen extension(s) return generators they don't want exhausted into memory?

Comment (maxhutch) [reply pif_ingestor/manager.py]: We can peek at the first element of the generator without collapsing everything into memory with [this pattern](https://stackoverflow.com/questions/2425270/how-to-look-ahead-one-element-in-a-python-generator)

**Comment by maxhutch** [comment]
We can easily get test coverage for generators in the ingest manger by changing the `test_csv_generator` to use automatic format detection (i.e. remove `-f template_csv` from `argv`).

**Pull request merged**